platforms/android/demo/src/main/java/com/mapzen/tangram/android/MainActivity.java

```diff
@@ -88,6 +189,63 @@ public void onLowMemory() {
         view.onLowMemory();
     }
 
+    @Override public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+        final ArrayList<SceneUpdate> sceneUpdates = new ArrayList<>(1);
+        final String apiKey = "vector-tiles-tyHL4AY";
+        sceneUpdates.add(new SceneUpdate("global.sdk_mapzen_api_key", apiKey));
+        spinner.setSelection(7);
+        map.loadSceneFile(sceneString.getText().toString(),sceneUpdates);
+    }
+
+    @Override public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+        if (map == null) { return; }
+
+        final ArrayList<SceneUpdate> sceneUpdates = new ArrayList<>(1);
+        final String apiKey = "vector-tiles-tyHL4AY";
+        sceneUpdates.add(new SceneUpdate("global.sdk_mapzen_api_key", apiKey));
+
+        switch (position) {
```

Comment:
since you do the same thing for all index values < scenes.length, you could refactor this to something like:

```
if (position < scenes.length) {
   sceneString.setText(scenes[position]);
   map.loadSceneFile(scenes[position], sceneUpdates);
} else {
   sceneString.setText(null);
   sceneString.requestFocus();
   InputMethodManager in = (InputMethodManager) getSystemService(Context.INPUT_METHOD_SERVICE);
   in.showSoftInput(sceneString, 0);
}
```

the else clause adds text clearing and shows the keyboard so the user can enter the custom scene location (if you don't want this behavior feel free to ignore though :) )